src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -2,20 +2,74 @@
 
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.MockitoJUnitRunner;
 
-import static org.junit.Assert.assertTrue;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
 public class ArrayListProductDaoTest
 {
-    private ProductDao productDao;
+    @Mock
+    private Product productWithoutPrice;
+
+    @Mock
+    private Product productOutOfStock;
+
+    @Mock
+    private Product rightProduct;
+
+    @Mock
+    private Product productForAdding;
+
+    @Spy
+    private List<Product> products = new ArrayList<>();
+
+    @InjectMocks
+    private ProductDao productDao = new ArrayListProductDao();
 
     @Before
-    public void setup() {
-        productDao = new ArrayListProductDao();
+    public void setup()
+    {
+        products.addAll(Arrays.asList(productWithoutPrice, productOutOfStock, rightProduct));
+
+        when(productWithoutPrice.getId()).thenReturn(1L);
+        when(productOutOfStock.getId()).thenReturn(2L);
+        when(productOutOfStock.getPrice()).thenReturn(new BigDecimal(1000));
+        when(rightProduct.getId()).thenReturn(3L);
+        when(rightProduct.getPrice()).thenReturn(new BigDecimal(1000));
+        when(rightProduct.getStock()).thenReturn(1000);
+        when(productForAdding.getId()).thenReturn(4L);
     }
 
     @Test
-    public void testFindProductsNoResults() {
-        assertTrue(productDao.findProducts().isEmpty());
+    public void testGetProduct()
+    {
+        assertEquals(productWithoutPrice, productDao.getProduct(1L));
+    }
+
+    @Test
+    public void testFindProducts()
+    {
+        assertEquals(1, productDao.findProducts().size());
+    }
+
+    @Test
+    public void testSaveAndDeleteProduct()
```

Comment:
Don't use a single test to verify logic of multiple methods. Make separate testSave and testDelete methods